src/Controller/GenreController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace App\Controller;
+
+use App\Entity\Genre;
+use App\Repository\GenreRepository;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Entity;
+use App\Form\GenreType;
+
+class GenreController extends BaseController
+{
+    const ITEMS_PER_PAGE = 10;
+
+    /**
+     * @Route("/admin/genre/list", name="list_genres")
+     * @Template("admin/genre/genre_list.html.twig")
+     */
+    public function listGenre(Request $request, GenreRepository $genreRepository)
+    {
+        $page = $request->query->get('page', 1);
```

Comment:
@daria-sieroshtan 
You can do just `$request->get('page', 1);`
It will search both `query` and `request`, as well as `attributes`